src/goto-programs/goto_convert.cpp

```diff
@@ -200,8 +194,9 @@ void goto_convertt::finish_computed_gotos(goto_programt &goto_program)
     goto_programt::instructiont &i=*g_it;
     exprt destination=i.code.op0();
 
-    assert(destination.id()==ID_dereference);
-    assert(destination.operands().size()==1);
+    DATA_INVARIANT(
+      destination.id() == ID_dereference, "dereference ID not allowed here");
```

Comment:
The message here seems to say the opposite of what's actually being asserted.

Reply:
`to_dereference_expr` should solve it nicely.